timemachine/cpp/src/hilbert_sort.cu

```diff
@@ -33,15 +33,14 @@ HilbertSort::HilbertSort(const int N)
 
     // estimate size needed to do radix sorting
     // reuse d_sort_keys_in_ rather than constructing a dummy output idxs buffer
-    cub::DeviceRadixSort::SortPairs(
+    gpuErrchk(cub::DeviceRadixSort::SortPairs(
```

Comment:
Interesting - I didn't know this returned a cudaError directly, does this kernel run synchronously?

Reply:
This call runs synchronously, but only to setup the size. If you provide all of the values, including a stream, it is async.